web/modules/custom/nys_dashboard/src/Form/ManageDashboardForm.php

```diff
@@ -0,0 +1,340 @@
+    ];
+
+    foreach ($this->fieldNamesToFlagTypes as $field_name => $flag_type) {
+      $options = $this->getUsersFlaggedEntitiesByFlagName($flag_type);
+      $uncheck_all_link = "<br><a href='' class='uncheck-all-link'>Uncheck all $field_name</a>";
+      $description = !empty($options)
+        ? "To stop following, uncheck $field_name and click Update My Preferences. $uncheck_all_link"
+        : "You're not following any $field_name.";
+      $form['followed_types_fieldset'][$field_name] = [
+        '#type' => 'checkboxes',
+        '#title' => ucfirst($field_name) . " You're Following",
+        '#description' => $description,
+        '#description_display' => 'before',
+        '#options' => $options,
+        '#default_value' => array_keys($options),
+      ];
+    }
+
+    $form['actions']['#type'] = 'actions';
+    $form['actions']['submit'] = [
+      '#type' => 'submit',
+      '#value' => $this->t('Update my preferences'),
+      '#button_type' => 'primary',
+      '#ajax' => [
+        'callback' => '::confirmFormModal',
+      ],
+    ];
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state): void {
+    // Populate $form_state 'content_to_unfollow'.
+    $content_to_unfollow = [];
+    $user_input = $form_state->getUserInput();
+    if (!empty($user_input)) {
+      foreach ($this->fieldNamesToFlagTypes as $field_name => $flag_type) {
+        // Break loop if user input empty, or if input doesn't match
+        // type_filter.
+        if (
+          empty($user_input[$field_name])
+          || $user_input['type_filter'] !== 'All' && $user_input['type_filter'] !== $field_name
+        ) {
+          continue;
+        }
+
+        $user_unchecked_entity_ids = array_diff(
+          array_keys($form['followed_types_fieldset'][$field_name]['#options']),
+          array_values($user_input[$field_name])
+        );
+        foreach ($user_unchecked_entity_ids as $flagged_entity_id) {
+          $content_to_unfollow[] = [
+            'flag_type' => $flag_type,
+            'flagged_entity_id' => $flagged_entity_id,
+            'field_name' => $field_name,
+          ];
+        }
+      }
+    }
+    $form_state->set('content_to_unfollow', $content_to_unfollow);
+
+    // Populate $form_state 'unfollow_content_ul'.
+    $unfollow_content_li = '';
+    foreach ($content_to_unfollow as $unfollow_content) {
+      $flagged_entity_type_id = ($unfollow_content['flag_type'] !== 'follow_this_bill') ? 'taxonomy_term' : 'node';
+      $flagged_entity = $this->entityTypeManager
+        ->getStorage($flagged_entity_type_id)
+        ->load($unfollow_content['flagged_entity_id']);
+      $content_type = substr($unfollow_content['field_name'], 0, -1);
+      $unfollow_content_li .= '<li>' . $content_type . ': '
+        . $flagged_entity->label() . '</li>';
+    }
+    $unfollow_content_ul = '<ul>' . $unfollow_content_li . '</ul>';
```

Comment:
@nathanielwoodland I imagine there is a reason for hardcoding HTML here. If possible, it would be good to use a list render array.

Reply:
Good catch! I'm sure I meant to move the HTML into twig. I bet forgot to put a todo above the code ...

Reply:
Made a wrong assumption about where this code was at first (I'd hardwired HTML in other areas prior to doing some theme/twig work, but that was all resolved previously), but still a good catch, and this has now been resolved @joelsteidl.